content/en/blog/docker-multi-arch/index.md

```diff
@@ -0,0 +1,114 @@
+1. **Multiple native nodes using the same builder instance**: With this approach, multiple hosts that run on different CPU architectures execute the build. The build time is faster. The drawback is that it requires access to native nodes.
+1. **Stage in a Dockerfile for cross-compilation**: This option is possible with languages that support cross-compilation. Arguments exposing the build and the target platforms are automatically available to the build stage. The build command can utilize these parameters to build the binary for the correct target.
+
+From these three options, we chose the first one, as it seemed the most straightforward route. However, in our case, the third option might have worked as well since we are building with tools that support cross-compilation, Rust, and GoLang. 
+
+A Docker CLI plugin, [buildx](https://docs.docker.com/buildx/working-with-buildx/), is required to build multi-architecture images. It extends the docker command with additional features, the multi-architecture build capability being one of them. Using buildx is almost the same as using the ordinary Docker build function. The target platform is added to the command wit the flag `--platform`.
+
+Example of building Docker image with buildx for arm64:
+```bash
+docker buildx build --platform linux/arm64 -t image_label .
+```
+
+## Building the Images
+
+Now we had chosen the strategy and had the tools installed. The next step was to review each image stack and ensure that it was possible to build all image layers for the needed variants.
+
+Our default image stack consists of [the custom base image](https://github.com/findy-network/findy-common-go/blob/master/infra/aws/Dockerfile.indy.ubuntu) and an application layer (service binary in question). The custom base image contains some tools and libraries that are common for all of our services. It expands the official Docker image for Ubuntu. For the official Docker images, there are no problems since Docker provides the needed variants out-of-the-shelf. However, our custom base image installs indy-sdk libraries from the Sovrin Debian repository, and unfortunately, the Debian repository did not provide binaries for arm64. So instead of installing the library from the Debian repository, we needed to add [a build step](https://github.com/findy-network/findy-common-go/blob/8bef1cbc4cc7d698275a69a9c9c4aff2622b84de/infra/aws/Dockerfile.indy.ubuntu#L12) that would build and install the indy-sdk from the sources.
+
+Otherwise, building for arm64 revealed no problems. The only needed change was that some of our server start scripts were missing [shebangs](https://en.wikipedia.org/wiki/Shebang_(Unix)) from the start of the script file. For yet an unknown reason, it prevented the shell from running the script properly on an arm64 container. The problem was fixed easily by adding the missing shebangs.
```

Comment:
I personally hate unknown reasons, so if there is any change we could find the reason by googling or something else it would be super!!!